bin/improver-nbhood-land-and-sea

```diff
@@ -0,0 +1,234 @@
+                       'the list of radii given first followed by a list of '
+                       'lead times to indicate at what lead time each radii '
+                       'should be used. For example: 10000,12000,14000 1,2,3 '
+                       'where a lead time of 1 hour uses a radius of 10000m, '
+                       'a lead time of 2 hours uses a radius of 12000m, etc.')
+
+    parser.add_argument('--ens_factor', metavar='ENS_FACTOR', type=float,
+                        default=1.0,
+                        help='The factor with which to adjust the '
+                        'neighbourhood size for more than one '
+                        'ensemble member. If ens_factor = 1.0 this '
+                        'essentially conserves ensemble members if '
+                        'every grid square is considered to be the '
+                        'equivalent of an ensemble member.'
+                        'Optional, defaults to 1.0.')
+    parser.add_argument('--sum_or_fraction', default="fraction",
+                        choices=["sum", "fraction"],
+                        help='The neighbourhood output can either be in the '
+                             'form of a sum of the neighbourhood, or a '
+                             'fraction calculated by dividing the sum of the '
+                             'neighbourhood by the neighbourhood area. '
+                             '"fraction" is the default option.')
+    parser.add_argument('--no_clip', action='store_true', default=False,
+                        help="By default the results of neighbourhooding will "
+                             "be clipped to the extremes in the input data. "
+                             "If '--no_clip' is used then this clipping is "
+                             "turned off. "
+                             "Once the neighbourhood code is fixed so it "
+                             "doesn't produce values above or below the "
+                             "extremes in the input data this option can be "
+                             "removed.")
+    parser.add_argument('--intermediate_filepath', default=None,
+                        help='If provided the result after neighbourhooding, '
+                             'before collapsing the extra dimension is saved '
+                             'in the given filepath.')
+
+    args = parser.parse_args()
+
+    cube = load_cube(args.input_filepath)
+
+    landmask = load_cube(args.input_mask_filepath)
+    # Create land and sea masks
+    land_only = landmask.copy(data=landmask.data.astype(int))
+    sea_only = landmask.copy(data=np.logical_not(landmask.data).astype(int))
+
+    if args.radius:
+        radius_or_radii = args.radius
+        lead_times = None
+    elif args.radii_by_lead_time:
+        radius_or_radii = args.radii_by_lead_time[0].split(",")
+        lead_times = args.radii_by_lead_time[1].split(",")
+
+    if args.additional_mask_filepath:
```

Comment:
There's a lot of code in this block, and I feel it needs proper unit tests!  Could this be implemented as a new plugin?  (As written here, but get the masking combination stuff captured outside the CLI.)

Reply:
This block has been simplified considerably. I have created bats tests to test the warnings/exceptions that are raised.